Blog with DRA updates in 1.33 (#49993)
kubernetes/website: content/en/blog/_posts/XXXX-XX-XX-dra-133-update.md

---
layout: blog
title: "Kubernetes 1.33: New features in DRA"
slug: dra-133-updates

> canonicalUrl: https://www.kubernetes.dev/blog/XXXX/XX/XX/dra-133-updates

Reviewer: Release comms will open a follow up PR that marks the PR not-draft and sets the correct publication date.

date: XXXX-XX-XX
author: >
  [Morten Torkildsen](https://github.com/mortent) (Google)

>   [Patrick Ohly](https://github.com/pohly) (Intel)

Reviewer 2: Hi, I'm an co-author of [KEP-5055: DRA: device taints and tolerations](https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/5055-dra-device-taints-and-tolerations/kep.yaml#L5).

Although I understand I'm not appropriate as an this blog author, I would be honored if I could get an credit in this blog as KEP-5055 co-author in some way.

And, how about putting credits for each KEP listed in this blog to honor/clarify contributors? We could perhaps list KEP authors/core contributors at the bottom in this blog?

What do you think??

Reviewer: We often do credit contributors to KEPs in post-release comms, usually right at the end of the article.

Author: I've added an "Acknowledgments" section at the end. I've just tried to keep it simple, with just names and github handles sorted alphabetically on the first name. But open to changing the format here.

I've used the names from slides for the [Device Management WG presentation](https://kccnceu2025.sched.com/event/1tczm/kubernetes-wg-device-management-gpus-tpus-nics-and-more-with-dra-kevin-klues-nvidia-patrick-ohly-intel) at Kubecon EU. I hope that is everyone, but happy to add more if I have forgotten anyone.

@pohly

---

> ## Kubernetes 1.33: New features in DRA

Reviewer: ```suggestion
```
We don't start articles with headings.

> Dynamic Resource Allocation (DRA) was originally introduced as an Alpha feature in 1.26, but went through a significant redesign for 1.31. The main DRA feature went to beta in 1.32 and is planned for GA in 1.34.

Reviewer: ```suggestion
Kubernetes [Dynamic Resource Allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/) (DRA) was originally introduced as an
Alpha feature in the 1.26 release, and then went through a significant redesign
for Kubernetes 1.31. The main DRA feature went to beta in 1.32 and the project hopes
it will be generally available in Kubernetes 1.34.
```

[Device Taints and Tolerations](https://github.com/kubernetes/enhancements/issues/5055) allow devices to be tainted and for workloads to tolerate those taints. This makes it possible for drivers or cluster administrators to mark devices as unavailable. Depending on the effect of the taint, this can prevent devices from being allocated or cause eviction of pods that are using the device.

[Prioritized List](https://github.com/kubernetes/enhancements/issues/4816) lets users specify a list of acceptable devices for their workloads, rather than just a single type of device. So while the workload might run best on a single high-performance GPU, it might also be able to run on 2 mid-level GPUs. The scheduler will attempt to satisfy the alternatives in the list in order, so the workload will be allocated the best set of devices available in the cluster.

> [Admin Access](https://github.com/kubernetes/enhancements/issues/5018) has been updated so that only users with access to a namespace with the `resource.k8s.io/admin-access: "true"` label are authorized to create ResourceClaim or ResourceClaimTemplates objects with the `adminAccess` field within the namespace. This grants administrators access to in-use devices and may enable additional permissions when making the device available in a container. This ensures that non-admin users cannot misuse the feature.

Reviewer: Please link to the documentation for these features rather than the KEPs.

(if the documentation isn't yet written, we may need to drop the enhancement from the release; we have a policy of only releasing user-facing features if they are documented)

Author: We have documentation for all the features mentioned, so I've updated the links to point to where the documentation will end up once published.

> ### Preparing for GA

Reviewer: ```suggestion
### Preparing for general availability
```

> A new v1beta2 API has been added to simplify the user experience and to prepare for additional features being added in the future. The RBAC rules for DRA have been improved and support has been added for seamless upgrades of DRA drivers.

Reviewer: ```suggestion
A new v1beta2 API has been added to simplify the user experience and to prepare for
additional features being added in the future. The RBAC rules for DRA
have been improved and support has been added for seamless upgrades of DRA drivers.
```

### What’s next?

> The plan for 1.34 is even more ambitious than for 1.33. Most importantly, we plan to bring DRA to GA, making it available by default on all Kubernetes clusters. This also means that all the DRA beta features will also be enabled by default, making it much easier to use them.

Reviewer: ```suggestion
The plan for 1.34 is even more ambitious than for 1.33. Most importantly, we
(the Kubernetes device management working group) hope to bring DRA to general availability, which will make it
available by default on all v1.34 Kubernetes clusters. This also means that many,
perhaps all, of the DRA features that are still beta in v1.34 will become enabled by default, making it much easier to use them.
```

[Device Taints and Tolerations](https://github.com/kubernetes/enhancements/issues/5055) allow devices to be tainted and for workloads to tolerate those taints. This makes it possible for drivers or cluster administrators to mark devices as unavailable. Depending on the effect of the taint, this can prevent devices from being allocated or cause eviction of pods that are using the device.

[Prioritized List](https://github.com/kubernetes/enhancements/issues/4816) lets users specify a list of acceptable devices for their workloads, rather than just a single type of device. So while the workload might run best on a single high-performance GPU, it might also be able to run on 2 mid-level GPUs. The scheduler will attempt to satisfy the alternatives in the list in order, so the workload will be allocated the best set of devices available in the cluster.

> [Admin Access](https://github.com/kubernetes/enhancements/issues/5018) has been updated so that only users with access to a namespace with the kubernetes.io/dra-admin-access label are authorized to create ResourceClaim or ResourceClaimTemplates objects within the namespace. This gives administrators privileged access to devices without compromising security.

Reviewer 3: ```suggestion
[Admin Access](https://github.com/kubernetes/enhancements/issues/5018) has been updated so that only users with access to a namespace with the `resource.k8s.io/admin-access: "true"` label are authorized to create ResourceClaim or ResourceClaimTemplates objects with the `adminAccess` field within the namespace. This grants administrators access to in-use devices and may enable additional permissions when making the device available in a container. This ensures that non-admin users cannot misuse the feature.
```

Author: Thanks @ritazh. I've updated the PR with your changes.

### What’s next?

The plan for 1.34 is even more ambitious than for 1.33. Most importantly, we (the Kubernetes device management working group) hope to bring DRA to general availability, which will make it available by default on all v1.34 Kubernetes clusters. This also means that many, perhaps all, of the DRA features that are still beta in v1.34 will become enabled by default, making it much easier to use them.

The Alpha features that were added in 1.33 will be brought to Beta in 1.34.

### Getting involved

A good starting point is joining the WG Device Management [Slack channel](https://kubernetes.slack.com/archives/C0409NGC1TK) and [meetings](https://docs.google.com/document/d/1qxI87VqGtgN7EAJlqVfxx86HGKEAc2A3SKru8nJHNkQ/edit?tab=t.0#heading=h.tgg8gganowxq) which happens at US/EU and EU/APAC friendly time slots.

Not all enhancement ideas are tracked as issues yet, so come talk to us if you want to help or have some ideas yourself! We have work to do at all levels, from difficult core changes to usability enhancements in kubectl which could be picked up by newcomers.

### Acknowledgments

A huge thanks to everyone who have contributed:

> * Cici Huang ([cici37](https://github.com/cici37))

Reviewer 4: ```suggestion
* Cici Huang ([cici37](https://github.com/cici37))
* Ed Bartosh ([bart0sh](https://github.com/bart0sh])
```

Ed has supported the "prepare for general availability" with various different PRs.

Author: Updated.

---
layout: blog

> title: "Kubernetes 1.33: New features in DRA"

Reviewer 5: nit:
```suggestion
title: "Kubernetes v1.33: New features in DRA"
```

Author: Thanks! I've updated the PR to use the `v` prefix for all mentions of Kubernetes versions and also use lowercase for all uses of `alpha` and `beta`. I also removed the capitalization of alpha and beta in the headers as the document doesn't otherwise use title case (and looking through previous blog post there doesn't seem to be a clear guidance on whether we should).

Reviewer 5: Thank you! I couldn't find a written convention for the feature stages either, but I see them mentioned in lowercase most times.

Reviewer 6: This was something we touched on in one of the SIG Docs or SIG Release meetings (I can't remember where it was), but I don't think it's clearly documented... This should be a part of some guideline indeed!

slug: dra-133-updates
draft: true
date: XXXX-XX-XX
author: >
  [Morten Torkildsen](https://github.com/mortent) (Google)
  [Patrick Ohly](https://github.com/pohly) (Intel)
---

> Kubernetes [Dynamic Resource Allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/) (DRA) was originally introduced as an Alpha feature in the 1.26 release, and then went through a significant redesign for Kubernetes 1.31. The main DRA feature went to beta in 1.32 and the project hopes it will be generally available in Kubernetes 1.34.

Reviewer 5: nit:
```suggestion
Kubernetes [Dynamic Resource Allocation](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/) (DRA) was originally introduced as an alpha feature in the v1.26 release, and then went through a significant redesign for Kubernetes v1.31. The main DRA feature went to beta in v1.32 and the project hopes it will be generally available in Kubernetes v1.34.
```

> The basic feature set of DRA provides a far more powerful and flexible API for requesting devices than Device Plugin. And while DRA remains a beta feature for 1.33, the DRA team has been hard at work implementing a number of new features and UX improvements. One feature has been promoted to Beta, while a number of new features have been added in Alpha. The team has also made progress towards getting DRA ready for GA.

Reviewer 5: ```suggestion
The basic feature set of DRA provides a far more powerful and flexible API for requesting devices than Device Plugin. And while DRA remains a beta feature for v1.33, the DRA team has been hard at work implementing a number of new features and UX improvements. One feature has been promoted to beta, while a number of new features have been added in alpha. The team has also made progress towards getting DRA ready for GA.
```

[Device Taints and Tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/dynamic-resource-allocation/#device-taints-and-tolerations) allow devices to be tainted and for workloads to tolerate those taints. This makes it possible for drivers or cluster administrators to mark devices as unavailable. Depending on the effect of the taint, this can prevent devices from being allocated or cause eviction of pods that are using the device.

[Prioritized List](https://kubernetes.io/docs/concepts/scheduling-eviction/dynamic-resource-allocation/#prioritized-list) lets users specify a list of acceptable devices for their workloads, rather than just a single type of device. So while the workload might run best on a single high-performance GPU, it might also be able to run on 2 mid-level GPUs. The scheduler will attempt to satisfy the alternatives in the list in order, so the workload will be allocated the best set of devices available in the cluster.

> [Admin Access](https://kubernetes.io/docs/concepts/scheduling-eviction/dynamic-resource-allocation/#admin-access) has been updated so that only users with access to a namespace with the `resource.k8s.io/admin-access: "true"` label are authorized to create ResourceClaim or ResourceClaimTemplates objects with the `adminAccess` field within the namespace. This grants administrators access to in-use devices and may enable additional permissions when making the device available in a container. This ensures that non-admin users cannot misuse the feature.

Reviewer 3: lgtm from sig auth

> [Admin Access](https://kubernetes.io/docs/concepts/scheduling-eviction/dynamic-resource-allocation/#admin-access) has been updated so that only users with access to a namespace with the `resource.k8s.io/admin-access: "true"` label are authorized to create ResourceClaim or ResourceClaimTemplates objects with the `adminAccess` field within the namespace. This grants administrators access to in-use devices and may enable additional permissions when making the device available in a container. This ensures that non-admin users cannot misuse the feature.

Reviewer 7: ```suggestion
[Admin Access](/docs/concepts/scheduling-eviction/dynamic-resource-allocation/#admin-access) has been updated so that only users with access to a namespace with the `resource.k8s.io/admin-access: "true"` label are authorized to create ResourceClaim or ResourceClaimTemplates objects with the `adminAccess` field within the namespace. This grants administrators access to in-use devices and may enable additional permissions when making the device available in a container. This ensures that non-admin users cannot misuse the feature.
```

### Preparing for general availability

A new v1beta2 API has been added to simplify the user experience and to prepare for additional features being added in the future. The RBAC rules for DRA have been improved and support has been added for seamless upgrades of DRA drivers.

### What’s next?

> The plan for 1.34 is even more ambitious than for 1.33. Most importantly, we (the Kubernetes device management working group) hope to bring DRA to general availability, which will make it available by default on all v1.34 Kubernetes clusters. This also means that many, perhaps all, of the DRA features that are still beta in v1.34 will become enabled by default, making it much easier to use them.

Reviewer 5: nit:
```suggestion
The plan for v1.34 is even more ambitious than for v1.33. Most importantly, we (the Kubernetes device management working group) hope to bring DRA to general availability, which will make it available by default on all v1.34 Kubernetes clusters. This also means that many, perhaps all, of the DRA features that are still beta in v1.34 will become enabled by default, making it much easier to use them.
```

> The Alpha features that were added in 1.33 will be brought to Beta in 1.34.

Reviewer 5: ```suggestion
The alpha features that were added in v1.33 will be brought to beta in v1.34.
```
